aes/aes-soft/src/fixslice.rs

```diff
@@ -0,0 +1,184 @@
+//! Fixsliced implementations of AES-128 and AES-256 (encryption-only)
+//! adapted from the C implementation.
+//!
+//! Fully-fixsliced implementation should run faster than the semi-fixsliced
+//! version at the cost of a bigger code size.
+//!
+//! See the paper at https://eprint.iacr.org/2020/1123.pdf for more details.
+//!
+//! # Author (original C code)
+//!
+//! Alexandre Adomnicai, Nanyang Technological University, Singapore
+//! <alexandre.adomnicai@ntu.edu.sg>
+
+// MIT License
+//
+// Copyright (c) 2020 Alexandre Adomnicai
```

Comment:
Note: I asked for permission to dual license the translation as Apache2/MIT and received it